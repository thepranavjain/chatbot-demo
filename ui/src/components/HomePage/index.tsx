import React, { useContext, useEffect } from "react";
import { GoogleAuthProvider, signInWithPopup } from "firebase/auth";
import { auth } from "../../firebase";
import { useNavigate } from "react-router-dom";
import { AuthContext } from "../../context/AuthContext";
import { ReactComponent as GoogleIcon } from "../../assets/icons/google-icon.svg";

const HomePage = () => {
  const navigate = useNavigate();
  const authContext = useContext(AuthContext);

  useEffect(() => {
    if (authContext?.user) {
      navigate("/chat");
    }

    return () => {};
  }, [authContext?.user, navigate]);

  const handleGoogleLogin = async () => {
    const provider = new GoogleAuthProvider();
    try {
      await signInWithPopup(auth, provider);
    } catch (error: any) {
      console.error("Error during login:", error.message);
    }
  };

  return (
    <div className="vh-100 vw-100 d-grid justify-content-center align-items-center">
      <div className="d-flex flex-column gap-3 align-items-center">
        <div>Welcome to my Chatbot app!</div>
        <div>Please login</div>
        <button
          onClick={handleGoogleLogin}
          className="btn btn-outline-primary d-grid justify-content-center align-items-center p-3"
        >
          <GoogleIcon height={"20px"} width={"20px"} />
        </button>
      </div>
    </div>
  );
};

export default HomePage;
