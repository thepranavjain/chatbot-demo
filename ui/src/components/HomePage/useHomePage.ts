import { useContext, useEffect } from "react";
import { GoogleAuthProvider, signInWithPopup } from "firebase/auth";
import { auth } from "../../firebase";
import { useNavigate } from "react-router-dom";
import { AuthContext } from "../../context/AuthContext";

const useHomePage = () => {
  const navigate = useNavigate();
  const authContext = useContext(AuthContext);

  useEffect(() => {
    if (authContext?.user) {
      navigate("/chat");
    }

    return () => {};
  }, [authContext?.user, navigate]);

  const handleGoogleLogin = async () => {
    const provider = new GoogleAuthProvider();
    try {
      await signInWithPopup(auth, provider);
    } catch (error: any) {
      console.error("Error during login:", error.message);
    }
  };

  return { handleGoogleLogin };
};

export default useHomePage;
