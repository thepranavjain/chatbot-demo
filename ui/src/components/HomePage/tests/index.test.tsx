import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import HomePage from "../index";
import useHomePage from "../useHomePage";

jest.mock("../useHomePage");

describe("HomePage", () => {
  const mockHandleGoogleLogin = jest.fn();

  beforeEach(() => {
    (useHomePage as jest.Mock).mockReturnValue({
      handleGoogleLogin: mockHandleGoogleLogin,
    });
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it("renders correctly", () => {
    const { asFragment } = render(<HomePage />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("calls handleGoogleLogin when the button is clicked", () => {
    render(<HomePage />);
    const loginButton = screen.getByRole("button");
    fireEvent.click(loginButton);
    expect(mockHandleGoogleLogin).toHaveBeenCalledTimes(1);
  });
});
