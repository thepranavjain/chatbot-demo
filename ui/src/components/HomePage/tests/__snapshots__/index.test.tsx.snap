// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage renders correctly 1`] = `
<DocumentFragment>
  <div
    class="vh-100 vw-100 d-grid justify-content-center align-items-center"
  >
    <div
      class="d-flex flex-column gap-3 align-items-center"
    >
      <div>
        Welcome to my Chatbot app!
      </div>
      <div>
        Please login
      </div>
      <button
        class="btn btn-outline-primary d-grid justify-content-center align-items-center p-3"
      >
        <svg
          height="20px"
          width="20px"
        >
          google-icon.svg
        </svg>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
